```
Error running mint automation
+ mkdir -p 9116-c0f99d8/gopath/src/github.com/minio
+ git -C 9116-c0f99d8/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 9116-c0f99d8/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 9116-c0f99d8/gopath/src/github.com/minio/minio fetch --quiet minio pull/9116/head:pr9116
+ git -C 9116-c0f99d8/gopath/src/github.com/minio/minio checkout --quiet pr9116
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/9116-c0f99d8/gopath
+ make -C 9116-c0f99d8/gopath/src/github.com/minio/minio --quiet
make: go: Command not found
make: go: Command not found
make: go: Command not found
make: go: Command not found
make: *** [checks] Error 1

```

--- a/cmd/api-headers.go
+++ b/cmd/api-headers.go
@@ -24,6 +24,7 @@ import (
 	"net/http"
 	"net/url"
 	"strconv"
+	"strings"
 	"time"
 
 	"github.com/minio/minio/cmd/crypto"
@@ -79,6 +80,10 @@ func setObjectHeaders(w http.ResponseWriter, objInfo ObjectInfo, rs *HTTPRangeSp
 	// Set Etag if available.
 	if objInfo.ETag != "" {
 		w.Header()[xhttp.ETag] = []string{"\"" + objInfo.ETag + "\""}
+	}
+
+	if strings.Contains(objInfo.ETag, "-") && len(objInfo.Parts) > 0 {
+		w.Header().Set(xhttp.AmzMpPartsCount, strconv.Itoa(len(objInfo.Parts)))
 	}
 
 	if objInfo.ContentType != "" {

--- a/cmd/http/headers.go
+++ b/cmd/http/headers.go
@@ -77,6 +77,9 @@ const (
 	AmzObjectLockLegalHold        = "X-Amz-Object-Lock-Legal-Hold"
 	AmzObjectLockBypassGovernance = "X-Amz-Bypass-Governance-Retention"
 
+	// Multipart parts count
+	AmzMpPartsCount = "x-amz-mp-parts-count"
+
 	// Dummy putBucketACL
 	AmzACL = "x-amz-acl"
 

